### ISSUE TITLE
Low precision & recall in WIDER FACE Dataset

### ISSUE THREAD

**Issue by FinnChou** [meituan/YOLOv6#25]
When I run train.py to training a face detector by WIDER FACE  dataset. 
the loss value is too large in epoch = 40,  and Average recall & Precision also not good. （see Training Log. ）

this dataset is the same format as Yolov5, and Yolov5 seems to work.
Am I missing something?

 
Training Log :  
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.003
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.009
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.002
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.005
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.003
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.030
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.076
Epoch: 40 | mAP@0.5: 0.0018804609699471495 | mAP@0.50:0.95: 0.0016776553250671712

     Epoch  iou_loss   l1_loss  obj_loss  cls_loss
    41/399     1.309    0.3974    0.8519    0.5262: 100%|██████████| 402/402 [02

**Comment by MTChengMeng** [comment]
> When I run train.py to training a face detector by WIDER FACE dataset. the loss value is too large in epoch = 40, and Average recall & Precision also not good. （see Training Log. ）
> 
> this dataset is the same format as Yolov5, and Yolov5 seems to work. Am I missing something?
> 
> Training Log : Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000 Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.003 Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.009 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.002 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.005 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.006 Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.003 Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.030 Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.076 Epoch: 40 | mAP@0.5: 0.0018804609699471495 | mAP@0.50:0.95: 0.0016776553250671712
> 
> ```
>  Epoch  iou_loss   l1_loss  obj_loss  cls_loss
> 41/399     1.309    0.3974    0.8519    0.5262: 100%|██████████| 402/402 [02
> ```
Thanks for your attention! The current version doesn't support all annotation formats. Could you provide your dataset.yaml?

**Comment by FinnChou** [comment]
> > When I run train.py to training a face detector by WIDER FACE dataset. the loss value is too large in epoch = 40, and Average recall & Precision also not good. （see Training Log. ）
> > this dataset is the same format as Yolov5, and Yolov5 seems to work. Am I missing something?
> > Training Log : Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000 Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.003 Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.009 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.002 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.005 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.006 Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.003 Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.030 Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.076 Epoch: 40 | mAP@0.5: 0.0018804609699471495 | mAP@0.50:0.95: 0.0016776553250671712
> > ```
> >  Epoch  iou_loss   l1_loss  obj_loss  cls_loss
> > 41/399     1.309    0.3974    0.8519    0.5262: 100%|██████████| 402/402 [02
> > ```
> 
> Thanks for your attention! The current version doesn't support all annotation formats. Could you provide your dataset.yaml?

thank you for your reply!  This is my dataset.yaml, but I think it should be ok.

```
# WiderFace FD dataset

train: '/data_ssd/WIDER_FACE_MAX_480_1Class/TrainSet/images/'
val: '/data_ssd/WIDER_FACE_MAX_480_1Class/TestSet/images/'
test: '/'

# number of classes
nc: 1

# class names
names: ['face']

```

And my label data has been converted from xyxy to cxcywh, such as

```
0 0.50146484375 0.2964860907759883 0.0361328125 0.0746705710102489 
0 0.4150390625 0.3323572474377745 0.037109375 0.07027818448023426 
0 0.3271484375 0.34260614934114203 0.037109375 0.06149341142020498 
0 0.234375 0.3616398243045388 0.037109375 0.06149341142020498 
```

**Comment by MTChengMeng** [comment]
> > > When I run train.py to training a face detector by WIDER FACE dataset. the loss value is too large in epoch = 40, and Average recall & Precision also not good. （see Training Log. ）
> > > this dataset is the same format as Yolov5, and Yolov5 seems to work. Am I missing something?
> > > Training Log : Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000 Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.003 Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.009 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.002 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.005 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.006 Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.003 Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.030 Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.076 Epoch: 40 | mAP@0.5: 0.0018804609699471495 | mAP@0.50:0.95: 0.0016776553250671712
> > > ```
> > >  Epoch  iou_loss   l1_loss  obj_loss  cls_loss
> > > 41/399     1.309    0.3974    0.8519    0.5262: 100%|██████████| 402/402 [02
> > > ```
> > 
> > 
> > Thanks for your attention! The current version doesn't support all annotation formats. Could you provide your dataset.yaml?
> 
> thank you for your reply! This is my dataset.yaml, but I think it should be ok.
> 
> ```
> # WiderFace FD dataset
> 
> train: '/data_ssd/WIDER_FACE_MAX_480_1Class/TrainSet/images/'
> val: '/data_ssd/WIDER_FACE_MAX_480_1Class/TestSet/images/'
> test: '/'
> 
> # number of classes
> nc: 1
> 
> # class names
> names: ['face']
> ```
> 
> And my label data has been converted from xyxy to cxcywh, such as
> 
> ```
> 0 0.50146484375 0.2964860907759883 0.0361328125 0.0746705710102489 
> 0 0.4150390625 0.3323572474377745 0.037109375 0.07027818448023426 
> 0 0.3271484375 0.34260614934114203 0.037109375 0.06149341142020498 
> 0 0.234375 0.3616398243045388 0.037109375 0.06149341142020498 
> ```

Maybe you could try removing the generated annotation file of coco format in '/data_ssd/WIDER_FACE_MAX_480_1Class/annotations', because there may be some wrong information cached in this file  before you successfully run the code.  And add '--check-images' and '--check-labels' options into your training script, then try again.
